includes/admin/reporting/export/class-batch-export-earnings-report.php

```diff
@@ -249,21 +249,12 @@ public function get_data() {
 		$data = array();
 
 		$start_date = date( 'Y-m-d 00:00:00', strtotime( $this->start ) );
-
-		if ( $this->count() == 0 ) {
-			$end_date = date( 'Y-m-d 23:59:59', strtotime( $this->end ) );
-		} else {
-			$end_date = date( 'Y-m-d 23:59:59', strtotime( 'first day of +1 month', strtotime( $start_date ) ) );
-		}
+		$end_date   = date( 'Y-m-t 23:59:59', strtotime( $this->start ) );
```

Comment:
```suggestion
		$end_date   = date( 'Y-m-t 23:59:59', strtotime( $this->end ) );
```

Reply:
`start` is actually correct here. (Unless you're seeing an error in the data?)

This `get_data()` method is meant to get records one month at a time. So if you select the range:

Jan 2021
to
March 2021

Then `$this->start` will be January 2021, and `$this->end` will be March 2021. However, `get_data()` only gets one month at a time. We're interested in data for the month of January only. So the local `$start_date` should be `2021-01-01 00:00:00` and the local `$end_date` should be `2021-01-31 23:59:59` -- not `2021-03-31 23:59:59`

Reply:
No, looking at it again, your way is getting correct data--I guess I am just confused and/or need more coffee.